src/figuresClasses.ts

```diff
@@ -1,11 +1,111 @@
-export interface Figure {}
+// export interface Figure {}
 
-export class Triangle implements Figure {}
+// export class Triangle implements Figure {}
 
-export class Circle implements Figure {}
+// export class Circle implements Figure {}
 
-export class Rectangle implements Figure {}
+// export class Rectangle implements Figure {}
 
-export function getInfo(figure): string {
-  return typeof figure;
+// export function getInfo(figure): string {
+//   return typeof figure;
+// }
+
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  private a: number;
+
+  private b: number;
+
+  private c: number;
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be greater than 0');
```

Comment:
[CODE STYLE] - If you throw an error try to provide its detailed description in error message. Instead of 'All sides must be greater than 0', you could specify which side is not greater than 0 to make the error message more informative.